CMakeLists.txt

```diff
@@ -0,0 +1,73 @@
+cmake_minimum_required(VERSION 3.20)
+
+project(sail_riscv)
+
+# Enable CTest
+enable_testing()
+
+# Export compile_commands.json for IDE support.
+set(CMAKE_EXPORT_COMPILE_COMMANDS TRUE)
+
+# Always use Position Independent Code. By default it is only used for
+# shared libraries (which require it), but you also need it for static
+# libraries if you link them into shared libraries.
+# Generally it just simplifies everything for a negligable performance cost.
+set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
+
+# Don't allow undefined symbols. This is generally a pain.
+include(CheckLinkerFlag)
+
+check_linker_flag(C "-Wl,--no-undefined" LINKER_SUPPORTS_NO_UNDEFINED)
+if (LINKER_SUPPORTS_NO_UNDEFINED)
+    add_link_options("-Wl,--no-undefined")
+endif()
+
+# Extra CMake files.
+set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules")
+
+# These are the main requirements.
+# Don't use `REQUIRED` so that we can print custom help messages.
+find_package(ZLIB)
+if (NOT ZLIB_FOUND)
+    message(FATAL_ERROR "Zlib not found. Try 'sudo apt install zlib1g-dev' or 'sudo dnf install zlib-devel'.")
```

Comment:
I'd bet that an inexperienced user on macOS will turn up, hit one of these messages and try to run the commands given, then complain that it doesn't work...

Reply:
Fair point, and actually we can easily customise the message for the OS - I'll do that.

Reply:
This is done - I added `brew` messages for mac, and don't suggest anything for windows.